src/services/session-manager.ts

```diff
@@ -60,6 +60,14 @@ export class OnlineSessionManager implements SessionManager {
   }
 
   private async update() {
+    if (process.env.REACT_APP_ALLOW_MANTENAINCE_AUTODETECT === 'true') {
+      try {
+        await this.dopplerLegacyClient.isDopplerMVCUp();
+      } catch {
+        this.updateSession({ status: 'maintenance' });
+        return;
+      }
+    }
```

Comment:
I don't understand what is the difference doing what the code below does and this. 🤔 
Isn't the same doing this with getUsetData?

Reply:
We tried this approach as you can see in the code, the issue is: when we call getUserData, we use CORS error to identify if user is not logged in, and when we tried reusing this call, we noticed that all calls to doppler from webapp returned CORS error when the server is offline. Explanation in spanish [here](https://github.com/MakingSense/Doppler/pull/7502)

Reply:
![image](https://user-images.githubusercontent.com/2439363/100117756-e55a5d80-2e53-11eb-9739-6b732648f2ff.png)

Reply:
Ok, maybe the other approach it could be useful when the user was already logged in the app but not for users that wants to entry for the first time. So it seems to be that this should work 👍

Reply:
For this to work we would need to change the actual behaviour for login, it's a bit risky.